File: Server (ASP.NET Core)/Petadmin/Models/OwnerPage.cs
using System.Collections.Generic;
using Petadmin.Core.Models;
using PetAdmin.Core.Models;

namespace Petadmin.Models
{
    public class OwnerPage
    {
        public Owner Owner { get; set; }
        public int OwnerTotalDebtAmount { get; set; }
        public List<Animal> AnimalsList { get; set; }
    }
}

File: Angular Client/models/owner-page.ts
import {Owner, OwnerEntity} from './owner';
import {Animal} from './animal';

export interface OwnerPage {
  owner: Owner;
  ownerTotalDebtAmount: number;
  animalsList: Animal[];
}

export class OwnerPageEntity implements OwnerPage {
  constructor() {
    this.animalsList = [];
    this.owner = new OwnerEntity();
    this.ownerTotalDebtAmount = 0;
  }
  animalsList: Animal[];
  owner: Owner;
  ownerTotalDebtAmount: number;
}
